docs/test/test.md

```diff
@@ -0,0 +1,80 @@
+
+# Introduction
+
+In order to provide high quality `pouch`, testing is an important part.
+
+This doc will give an introduction about the following three parts:
+
+*  the organization of pouch test
+*  the usage of pouch test
+*  the development of pouch test
+
+# Organization of test
+
+Test in pouch could be divided into two parts [unit testing](https://en.wikipedia.org/wiki/Unit_testing#Description) and [integration testing](https://en.wikipedia.org/wiki/Integration_testing). More info could be get from the wiki page. For pouch developer, if your code is only used in a single module, then the unit test is enough. While if your code is called by multiple modules, integration tests are required. In pouch, both of them are developed using go language. More details could be got in [Unit Testing](#unit-test) and [Integration Testing](#intg-test).
+
+# Usage of Test
+
+## <a name="unit-test"></a> Unit Testing
+
+Unit testing uses [go testing](https://golang.org/pkg/testing/) package, named with `_test.go` postfix and always locates in the same directory with the code tested. `client/client_test.go` is a good example of unit test.
+
+There are two ways to trigger unit test, from the makefile target to run entire unit test or just using `go test $testdir` to run unit test in a specified directory. Here is an example:
```

Comment:
s/makefile/Makefile
and also provide a hyperlink for work `Makefile`:

There are two ways to trigger unit test. First, Use [Makefile](https://github.com/alibaba/pouch/blob/master/Makefile) target `unit-test` to run entire unit test, and the command is `make unit-test` in the root path of pouch. Second, use `go test $testdir` to run unit test in a specified directory. Here is an example: